js/values/AssessmentResult.js

```diff
@@ -17,6 +17,7 @@ var emptyMarker = function() {
 var AssessmentResult = function() {
 	this._hasScore = false;
 	this._identifier = "";
+	this._shouldMark = false;
```

Comment:
I think we should to with `_hasMarks` instead of `_shouldMark`. You have already intuitively named the variable in the assessment `hasMarks` so it feels right to name it that way everywhere.